packages/langium/src/lsp/workspace-symbol-provider.ts

```diff
@@ -46,11 +54,12 @@ export class DefaultWorkspaceSymbolProvider implements WorkspaceSymbolProvider {
         this.fuzzyMatcher = services.lsp.FuzzyMatcher;
     }
 
-    async getSymbols(params: WorkspaceSymbolParams, cancelToken = CancellationToken.None): Promise<WorkspaceSymbol[]> {
+    async getSymbols(params: WorkspaceSymbolParams, cancelToken?: CancellationToken): Promise<WorkspaceSymbol[]> {
+        const token = cancelToken ?? CancellationToken.None;
```

Comment:
See also https://github.com/eclipse-langium/langium/pull/1605#discussion_r1696957672. Maybe it's a matter of taste, but I don't see this as an improvement w.r.t. readability.

My argument for keeping the original syntax: By specifying the default value as part of the method signature, it's like a part of the API and thus more transparent for those using it.

Reply:
The modified solution also introduces an additional symbol in the scope so now there's `token` and `cancelToken`. This is a potential source of error, as the difference is easy to overlook. It can be mitigated by overriding the parameter value instead of adding another const variable. But even then, it's an additional line of code that needs to be understood, so I prefer the more concise form.

Reply:
Reverted default parameters.